### DISCUSSION TITLE
Konstanten per xml überschreibbar

### DISCUSSION THREAD

**Discussion by rohne83** [Courseplay/Courseplay_FS22#2374] [Ideas]
Aktuell sind ja die meisten der ganzen wichtigen Zahlenwerte (zB 80% Drescherfüllstand wann Abfahrer gerufen werden soll, 10% ULW-Füllstand bis zu dem nach neuem Trailer gesucht werden soll, wohl auch der Siloradius beim Siloschieben) meist hartcodierte Zahlen irgendwo in den LUAs. Würde es nicht Sinn machen, diese als Variablen (bzw hier wären es eher Konstanten) möglichst global zu definieren? Dürfte einerseits die Wartbarkeit des Codes verbessern, und böte die Möglichkeit diese Werte auch ähnlich wie mit den vehicleConfigurations per xml zu "manipulieren". Wenn diese dann ebenfalls ähnlich zu cpReadVehicleConfigurations ingame aktualiserbar wäre, könnte das sicherlich auch das Testen während der Entwicklung deutlich bequemer machen.
Netter Nebeneffekt für die Nutzer: zumindest die erfahrenen Nutzer hätten wieder die Möglichkeit diese Werte an ihre speziellen Erfordernisse anzupassen, ohne dass dafür zusätzliche Settings im ESC-Menü oder im Hud notwendig wären ;)

**Comment by Tensuko** [comment]
Es ist absicht, dass die Zahlen so gewählt sind und die werden wir auch nicht veränderbar machen.
Auch nicht per XML, da dort noch mehr "kaputt" gemacht werden kann bei falscher Eingabe.

Grund: Stabilität.
Anpassbare Werte durch User haben immer mehr Auswirkungen als man glaubt und gerade sowas verstecktes wird dann doch auch früher oder später von unerfahrenen Usern verwendet die dann mit Problemen ankommen und man nicht weis was die getrieben haben...

Außerdem ist es nicht in unserem Interesse so viele anpassbare Settings zu haben, bereits im 19er gab es zu viel was immer dazu führt, dass man Usern erklären musste was man wofür wann wie einstellen muss.

Leider muss ich das Beispiel AD und Kolission nennen.... wäre für mich ein NO-GO.
